### PR TITLE
Add dependency_overrides to purchases_flutter_ui

### PR DESCRIPTION
**Pull request by vegaro**
I wasn't sure what was going on in #995 but there was some issues importing a newly added model. The problem is that we were depending on version ^6.0.0 of purchases_flutter. 

I made some changes to the pubspec.yaml, so now we use the local plugin. 

According to [the docs](https://dart.dev/tools/pub/dependencies), "As a result, if you publish a package to pub.dev, keep in mind that your package's dependency overrides are ignored by all users of your package.". So we should be fine to publish it like this.

### CODE REVIEW

File: purchases_ui_flutter/pubspec.yaml [hunk 1]
```diff
@@ -11,14 +11,15 @@ environment:
...
-  # We can't depend on exactly the same version
-  # because it won't be available when bumping the version
-  # until it's released.
-  purchases_flutter: ^6.0.0
+  purchases_flutter: ^6.22.0
   flutter:
     sdk: flutter
   plugin_platform_interface: ^2.0.2
 
+dependency_overrides:
```
Comment (tonidero) [inline purchases_ui_flutter/pubspec.yaml:19]: Maybe add a comment explaining this shouldn't affect published package?

**Review by tonidero — APPROVED** [review]

File: fastlane/Fastfile [hunk 2]
```diff
@@ -25,7 +25,7 @@ files_with_version_number = {
     './ios/Classes/PurchasesFlutterPlugin.m' => ['return @"{x}"'],
     './android/build.gradle' => ['version \'{x}\''],
     './android/src/main/java/com/revenuecat/purchases_flutter/PurchasesFlutterPlugin.java' => ['PLUGIN_VERSION = "{x}"'],
-    './purchases_ui_flutter/pubspec.yaml' => ['version: {x}'],
+    './purchases_ui_flutter/pubspec.yaml' => ['version: {x}', 'purchases_flutter: ^{x}'],
```
Comment (tonidero) [inline fastlane/Fastfile:28]: I guess if we are doing this, we could specify the specific package (which could probably be a good thing IMO...

**Review by tonidero — APPROVED** [review]

File: fastlane/Fastfile (hunk 2 above)
Comment (vegaro) [reply fastlane/Fastfile:28]: do you mean the specific package version? I got a warning when doing that:
```
* Your dependency on "purchases_flutter" should allow more than one version. For example:
  
  dependencies:
    purchases_flutter: ^6.22.0
  
  Constraints that are too tight will make it difficult for people to use your package
  along with other packages that also depend on "purchases_flutter".
```

should I add the specific version and ignore that?

Comment (tonidero) [reply fastlane/Fastfile]: Oh sorry! Yeah I meant that... Basically so people don't accidentally use a mismatched version of purchases_flutter and purchases_ui_flutter

Comment (vegaro) [reply fastlane/Fastfile]: perfect I did that. Makes sense. We will get a warning when publishing but it's fine

**Pull request merged**